tests/unit/kafka_connection_test.cc

```diff
@@ -0,0 +1,110 @@
+/*
+ * This file is open source software, licensed to you under the terms
+ * of the Apache License, Version 2.0 (the "License").  See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.  You may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*
+ * Copyright (C) 2019 ScyllaDB Ltd.
+ */
+
+#include <seastar/testing/test_case.hh>
+#include <seastar/testing/test_runner.hh>
+#include "../../src/kafka/connection/tcp_connection.hh"
+
+using namespace seastar;
+
+// All of the tests below assume that there is a Kafka broker running
+// on address BROKER_ADDRESS
+constexpr char BROKER_ADDRESS[] = "172.13.0.1:9092";
+
+constexpr char message_str[] = "\x00\x00\x00\x0E\x00\x12\x00\x02\x00\x00\x00\x00\x00\x04\x74\x65\x73\x74";
+constexpr size_t message_len = 18;
+
+SEASTAR_TEST_CASE(kafka_establish_connection_test) {
```

Comment:
if you use SEASTAR_THREAD_TEST_CASE then you will be able to simplify the tests and just wait for each operation to finish with .get() or .get0().

Reply:
Makes sense. I'd include this in the later rework separating and expanding the connection tests, unless you think it's important for this PR.

Reply:
It's good to get it right from the start but I won't be stubborn about it.